src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,272 @@
+
+                        try {
+                            Method getterMethod = entityClass.getMethod(getterName);
+                            Object actualValue = getterMethod.invoke(entity);
+
+                            if (!expectedValue.equals(actualValue)) {
+                                return false;
+                            }
+                        } catch (IllegalAccessException e) {
+                            logger.error("Unable to access property " + fieldName);
+                            e.printStackTrace();
+                            return false;
+                        } catch (InvocationTargetException | NoSuchMethodException e) {
+                            logger.error("No such method for " + fieldName);
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    return true;
+                })
+                .collect(Collectors.toList());
+    }
+
+    public String getDatabasePath(Class<? extends BaseEntity> entityClass) {
+        var databaseFolder = settings.getDatabasePath();
+        return databaseFolder + File.separator + entityClass.getSimpleName() + "Table" + ".json";
+    }
+
+    private <T extends BaseEntity> void updateEntityFields(T outcomeEntity, T incomeEntity) {
+        Field[] fields = incomeEntity.getClass().getDeclaredFields();
+        for (Field field : fields) {
+            String fieldName = field.getName();
+            String setterName = "set" + Character.toUpperCase(fieldName.charAt(0)) + fieldName.substring(1);
+            String getterName = getGetterName(incomeEntity.getClass(), fieldName);
+
+            try {
+                Method getterMethod = incomeEntity.getClass().getMethod(getterName);
+                Object value = getterMethod.invoke(incomeEntity);
+
+                Method setterMethod = outcomeEntity.getClass().getMethod(setterName, field.getType());
+                setterMethod.invoke(outcomeEntity, value);
+            } catch (IllegalAccessException | NoSuchMethodException e) {
+                logger.error("Unable to access property " + fieldName);
+                e.printStackTrace();
+            } catch (InvocationTargetException e) {
+                logger.error("Invocation Target Exception for " + fieldName);
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private <T extends BaseEntity> List<T> deserializeEntities(Class<? extends BaseEntity> entityClass,
+                                                               String content) {
+        var typeFactory = objectMapper.getTypeFactory();
+        var genericType = typeFactory.constructType(entityClass);
+        var listType = typeFactory.constructCollectionType(List.class, genericType);
+
+        try {
+            return objectMapper.readValue(content, listType);
+        } catch (IOException e) {
+            logger.error("Unable to deserialize data at " + content);
```

Comment:
Our database can be large, consider not adding all the data to log file. Just add entity class  as additional information.